### ISSUE TITLE
stan_lm error if model has no intercept

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstanarm#66]
@bgoodri Not that it's a particularly good idea to use `stan_lm` without an intercept, but it doesn't seem to work: 

```{r}
stan_lm(mpg ~ -1 + ., data = mtcars, prior = R2(0.5))

SAMPLING FOR MODEL 'lm' NOW (CHAIN 1).
[1] "Error : Initialization partially from source failed."
[2] "variable z_alpha missing"                            
error occurred during calling the sampler; sampling not done
```

**Comment by bgoodri** [comment]
OK, I'll fix it.

**Comment by bgoodri** [comment]
This was fixed by 29b60cfc16add7fd5e999692267a700e24ae44c7
